cpp/src/structure/graph.cu

```diff
@@ -28,7 +61,61 @@ void GraphCompressedSparseBase<VT,ET,WT>::get_source_indices(VT *src_indices) co
   cugraph::detail::offsets_to_indices<VT>(offsets, GraphBase<VT,ET,WT>::number_of_vertices, src_indices);
 }
 
+template <typename VT, typename ET, typename WT>
+void GraphCOO<VT,ET,WT>::degree(ET *degree, int x) const {
+  //
+  // NOTE:  We assume offsets/indices are a CSR.  If a CSC is passed
+  //        in then x should be modified to reflect the expected direction.
+  //        (e.g. if you have a CSC and you want in-degree (x=1) then pass
+  //        the offsets/indices and request an out-degree (x=2))
+  //
+
+  // Calculates the degree of all vertices of the graph
+  // x = 0: in+out degree
+  // x = 1: in-degree
+  // x = 2: out-degree
+
+  cudaStream_t stream{nullptr};
+
+  if (x != 1) {
+    degree_from_vertex_ids(GraphBase<VT,ET,WT>::number_of_edges, src_indices, degree, stream);
+  }
+
+  if (x != 2) {
+    degree_from_vertex_ids(GraphBase<VT,ET,WT>::number_of_edges, dst_indices, degree, stream);
+  }
+}
+
+template <typename VT, typename ET, typename WT>
+void GraphCompressedSparseBase<VT,ET,WT>::degree(ET *degree, int x) const {
```

Comment:
Might be overkill but would it make sense to make x an enum?

Reply:
We do that in other cases, it would be a little clearer.  I've mostly been keeping the interfaces the same as I work through the gdf_column changes except for the gdf_column portions of the interface.

This might be a  reasonable place to make a further change to the C++ API.  Opinions?  @afender , @BradReesWork?

Reply:
If you can quickly create it as an enum, that would be great.  Else you can add the /** TO D0 **/ tag and it will be added to our tech debt